lib/errors.js

```diff
@@ -1,39 +1,73 @@
+'use strict';
+
+var util = require('util');
+
+/**
+ * The base class for all thinky errors
+ *
+ * @constructor
+ * @alias Error
+ */
+var ThinkyError = function(message) {
```

Comment:
I think @neumino asking why not continue using function declarations (i.e. `function ThinkyError(message) { ... }`) instead? Either way `ThinkyError` will be defined in this scope so I'm not sure what you mean by easily reused/exported.